src/extension/messages.ts

```diff
@@ -1,16 +1,41 @@
 import { ExplorerResponse } from "../types";
 import { GetStates, GetContext } from "../application/stateMachine";
 import { DevtoolsMachine } from "../application/machines";
+import { RPCMessage } from "./rpc";
 
 export interface MessageFormat {
   type: string;
+  [key: string]: unknown;
 }
 
-export type ApolloClientDevtoolsMessage<Message extends MessageFormat> = {
+export const enum MessageType {
+  RPC = "rpc",
+  Event = "event",
```

Comment:
I'm not super in love with "event" as the name since this might get confusing with the browser built-in `Event` object. These are meant as fire-and-forget messages send through `actor`. [Worker RPC](https://github.com/DirtyHairy/worker-rpc) calls these "signals". Perhaps that makes more sense?